### ISSUE TITLE
Remove temp file logging for each response

### ISSUE THREAD

**Issue by amoose** [department-of-veterans-affairs/connect_vbms#101]
The [java decryption script](https://github.com/department-of-veterans-affairs/connect_vbms/blob/master/src/do_wsse.sh#L53) is writing the decryption output to a temp file each time a response is received. 

This is useful for debugging, but needs to configurable and disabled by default.

**Comment by llimllib** [comment]
I inaccurately identified the code that is pointed to above as the place where the logging occurs; it's not. It actually occurs via [wsse setting the log file path](https://github.com/department-of-veterans-affairs/connect_vbms/blob/master/src/do_wsse.sh#L53) for log4j, which logs via one of the java libs in some crazy way that only java people will understand.

The code pointed to above just creates the temporary log file name.

**Comment by amoose** [comment]
Thanks for that clarification, @llimllib. updated.

**Comment by harrisj** [comment]
Am I dumb or could we just handle this for now by setting the logfile path to be /dev/null?

**Comment by llimllib** [comment]
We could set it to that in `do_wsse.sh`, but then nobody would have any logging; also we'd still be keeping around open temporary file handles so this bug wouldn't be closed. Also any potential windows use would go away? Though I have no idea if there is that potential.

(But monkeypatching to that might be a quick fix for the log-files-in-production issue that sparked this issue)

**Comment by harrisj** [comment]
Okay, so there are two places where the WSSE script seems to be specifying a logfile

Uploading appends to a single log file in the script root
```sh
if [ "$MODE" = EncryptSOAPDocument ]; then
  [ -z "$REQNAME" ] && echo "Specify request name in -n" >&2 && exit 1
  ARG="$REQNAME"
  LOGFILE="${SCRIPT_ROOT}/log/upload.log"
elif [ "$MODE" = DecryptMessage ]; then
  [ -z "$LOGFILE" ] && echo "Specify outfile with -l" >&2 && exit 1
else
  echo "Unknown Mode...how did that happen??" >&2
  exit 1
fi
```

while debugging is set in the call to `do_wsse.sh`

```ruby
data = Tempfile.open('log') do |out_t|
    VBMS.decrypt_message_xml(body, @keyfile, @keypass, out_t.path)
end
```

It seems like we have several options:
1. We can disable the logging entirely. This would mean removing the Tempfile creation and commenting out the `-dlogfilname` argument invocation in `do_wsse.sh`
2. We could pick a single file for the decryption logging. The problem with the current approach is that even if we don't want the logging, it's still creating temp files and eating up inodes

It's worth noting that all this logging will go away entirely once we switch to Ruby encryption, so this measure is only for the short term. Thoughts?

**Comment by alex** [comment]
Personally I have no problem dropping the logging entirely.

On Mon, Nov 9, 2015 at 10:11 PM, Jacob Harris <notifications@github.com>
wrote:

> Okay, so there are two places where the WSSE script seems to be specifying
> a logfile
>
> Uploading appends to a single log file in the script root
>
> if [ "$MODE" = EncryptSOAPDocument ]; then
>   [ -z "$REQNAME" ] && echo "Specify request name in -n" >&2 && exit 1
>   ARG="$REQNAME"
>   LOGFILE="${SCRIPT_ROOT}/log/upload.log"elif [ "$MODE" = DecryptMessage ]; then
>   [ -z "$LOGFILE" ] && echo "Specify outfile with -l" >&2 && exit 1else
>   echo "Unknown Mode...how did that happen??" >&2
>   exit 1fi
>
> while debugging is set in the call to do_wsse.sh
>
> data = Tempfile.open('log') do |out_t|
>     VBMS.decrypt_message_xml(body, @keyfile, @keypass, out_t.path)end
>
> It seems like we have several options:
> 1. We can disable the logging entirely. This would mean removing the
> Tempfile creation and commenting out the -dlogfilname argument invocation
> in do_wsse.sh
> 2. We could pick a single file for the decryption logging. The problem
> with the current approach is that even if we don't want the logging, it's
> still creating temp files and eating up inodes
>
> It's worth noting that all this logging will go away entirely once we
> switch to Ruby encryption, so this measure is only for the short term.
> Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/department-of-veterans-affairs/connect_vbms/issues/101#issuecomment-155269977>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

**Comment by llimllib** [comment]
ditto, it seems like final remnants of my dark days of debugging

**Comment by harrisj** [comment]
This has been handled by PR #106 which superseded the fixes in #103 and #104
